spanner/spec/spec_helper.rb

```diff
@@ -247,4 +249,17 @@ def restore_database_from_backup
 
     @test_database
   end
+
+  def with_retry action, attempts: 5
+    @attempt_number ||= 0
```

Comment:
Since `@attempt_number`'s squared value is used for sleep,  I suggest having a max cap on this value (say, 10). Else if a user of this helper passes a large number without being aware of the exponential increase, the CI will go on for a long time.

Reply:
Actually, I just remembered solving a similar issue in the past for storage ([link](https://github.com/googleapis/google-cloud-ruby/blob/c1d11a9544a422dfd3e03c1d1ed21e929ec181eb/google-cloud-storage/acceptance/storage_helper.rb#L109)). You can optionally consider using the `Retriable` gem so we don't have to deal with the custom logic.

Reply:
ya that makes sense, have tried updating to use Retriable.